src/opencl.nim

```diff
@@ -188,9 +188,8 @@ type
 proc raiseEOpenCL*(x: TClResult) {.noinline.} =
   raise newException(EOpenCL, $x & " " & $int(x))
 
-template check*(a: expr) =
-  let y = a

```

Comment:
There is a reason for this ``let`` statement.

Reply:
Ah I see, the side-effects.
So would this be fine?

```Nim
template check*(a: TClResult) =
  let y = a
  if y != TClResult.Success: raiseEOpenCL(y)
```

Reply:
yes